analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/ExecutingSqlQueriesBase.cs

```diff
@@ -129,18 +138,46 @@ protected override void Initialize(SonarAnalysisContext context)
             PropertyAccessTracker.Track(context,
                 PropertyAccessTracker.MatchProperty(properties),
                 PropertyAccessTracker.MatchSetter(),
-                c => IsTracked(GetSetValue(c), c.SemanticModel),
+                c => IsTracked(GetSetValue(c), c),
                 Conditions.ExceptWhen(PropertyAccessTracker.AssignedValueIsConstant()));
 
             TrackObjectCreation(context, constructorsForFirstArgument, FirstArgumentIndex);
             TrackObjectCreation(context, constructorsForSecondArgument, SecondArgumentIndex);
         }
 
+        protected bool IsTrackedVariableDeclaration(TExpressionSyntax expression, SyntaxBaseContext context)
+        {
+            var node = expression;
+            while (node is TIdentifierNameSyntax identifierNameSyntax)
+            {
+                var identifierName = GetIdentifierName(identifierNameSyntax);
+                node = AssignmentFinder.FindLinearPrecedingAssignmentExpression(identifierName, node) as TExpressionSyntax;
+
+                if (IsSensitiveExpression(node, context.SemanticModel))
+                {
+                    context.AddSecondaryLocation(
```

Comment:
This is long and repetitive. Can you craft the `SecondaryLocation` instance inside `AddSecondaryLocation`? Something like `AddSecondaryLocation(Location location, String message, params String[] formatArgs)`